contextIO.go

```diff
@@ -95,6 +100,23 @@ func (c *ContextIO) NewRequest(method, q string, queryParams url.Values, body io
 	return req, nil
 }
 
+func (c *ContextIO) wait() {
+	if c.rate > 0 && c.lastRequest.Unix() > 0 {
+		sinceLast := time.Now().Sub(c.lastRequest)
+		sleep := c.rate - sinceLast
+		if sleep > 0 {
+			time.Sleep(sleep)
+		}
+	}
+	c.lastRequest = time.Now()
+}
+
+// SetRate the number of possible requests per minute
+func (c *ContextIO) SetRate(requestsPerMinute float64) {
```

Comment:
I like this.  There is a rate limit in C.IO on requests that end up making IMAP calls to avoid being throttled by the IMAP servers.  Currently it is 30/minute on 2.0 and 240/min on Lite.  Context.IO is working on adding this info to the docs, and those limits don't apply on requests that are being pulled from our local cache, which is most requests on 2.0 and user info on Lite.   I would like for a sensible default on this value though, so you don't have to set the rate all the time (that would break previous behavior on the lib).  I think the lite limit would be fine (240/min) since very few of the 2.0 calls go to IMAP (just certain headers, email bodies).